goodai/ltm/keywords/keywords.py

```diff
@@ -0,0 +1,75 @@
+import spacy
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+# Load the English model
+nlp = spacy.load("en_core_web_sm")
+
+# Check if a keyword is valid
+def is_valid_keyword(keyword: str) -> bool:
+    keyword_doc = nlp(keyword)
```

Comment:
I might be doing early optimization, but I've measured an average minimum of 1.8 ms for each call to `nlp`, while it's around 5 ms to run it on a larger text. The avg cost of running `extract_keywords` is 78 ms, which might be attributed to the `nlp` nested calls. It might be worth trying passing the `Span` objects, if possible.

Reply:
I did some profiling and lines 27-37 account for 89% of the time.